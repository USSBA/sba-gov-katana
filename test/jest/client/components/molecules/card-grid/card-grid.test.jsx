/*global expect*/

import React from 'react';
import CardGrid from 'client/components/molecules/card-grid/card-grid.jsx';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import _ from "lodash";

test('CardGrid renders a simple 3x3 grid', () => {
  let cards = _.map(Array(9), (item, index)=> {return {text: "this is card #"+index} })
  let renderCard = (data, index)=>{(<p>{data.text}</p>)}
  const component = renderer.create(<CardGrid cards={cards} renderCard={renderCard}/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
