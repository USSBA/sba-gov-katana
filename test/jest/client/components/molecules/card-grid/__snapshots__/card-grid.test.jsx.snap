// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardGrid renders a simple 3x3 grid 1`] = `
<div
  className="card-grid container"
>
  <div
    className="cardRow"
  >
    <div
      className="card-container card"
    />
    <div
      className="card-container card"
    />
    <div
      className="card-container card"
    />
  </div>
  <div
    className="cardRow"
  >
    <div
      className="card-container card"
    />
    <div
      className="card-container card"
    />
    <div
      className="card-container card"
    />
  </div>
  <div
    className="cardRow"
  >
    <div
      className="card-container card"
    />
    <div
      className="card-container card"
    />
    <div
      className="card-container card"
    />
  </div>
</div>
`;
